src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -306,6 +306,10 @@ public static function createFromGlobals()
         ) {
             parse_str($request->getContent(), $data);
             $request->request = new ParameterBag($data);
+        } elseif (in_array($request->headers->get('CONTENT_TYPE'), array('application/json', 'application/x-json')) &&
+            in_array(strtoupper($request->server->get('REQUEST_METHOD', 'GET')), array('POST', 'PUT', 'DELETE', 'PATCH'))
+        ) {
+            $request->request = new ParameterBag(json_decode($request->getContent(), true));
```

Comment:
If the JSON is invalid, `json_decode` returns `false`. You should check this, also you should check if the decoded json is an array.